docs/creating-a-static-check.md

```diff
@@ -0,0 +1,165 @@
+# Creating a Static Check
+
+Static checks are used to perform static analysis against a plugin's codebase.
+
+The Static checks, which analyze the code, either using PHP CodeSniffer sniffs or custom logic e.g. using regular expressions.
+
+```php
+use WordPress\Plugin_Check\Checker\Check_Result;
+use WordPress\Plugin_Check\Checker\Check_Categories;
+use WordPress\Plugin_Check\Checker\Static_Check;
+use WordPress\Plugin_Check\Traits\Stable_Check;
+
+class Custom_Check extends Static_Check {
+
+  use Stable_Check;
+  
+  public function get_categories() {
+    // Return an array of check categories.
+  }
+
+  public function run( Check_Result $result );
+    // Handle running the check and adding warnings or errors to the result.
+  }
+}
+```
+
+Here are two more concrete ways to write static checks.
+
+## Creating a new Check using PHP CodeSniffer.
+
+A new Check class should be created for the static check and should implement the following attributes.
+
+- The class name should be suffixed with `_Check`.
+- The class should extend the `Abstract_PHP_CodeSniffer_Check` class.
+- The class should implement the `get_args()` method.
+
+## Creating a new Check using Abstract_File_Check.
+
+A new Check class should be created for the static check and should implement the following attributes.
+
+- The class name should be suffixed with `_Check`.
+- The class should extend the `Abstract_File_Check` class.
+- The class should implement the `check_files()` method.
+
+### PHPCS Arguments
+
+Rather than write all the logic required to run PHPCS, the `Abstract_PHP_CodeSniffer_Check` handles this logic so developers only need to supply the arguements to PHPCS in order to get their check working.
+
+This is done by implementing the `get_args()` method in the new Check class defined within the `Abstract_PHP_CodeSniffer_Check`.
+
+The `get_args()` method should return an associative array containing the PHPCS arguments including the file extension, code standard and specific sniff to run.
+
+Below is an example of a Static Check class that checks for i18n usage in the plugins codebase.
+
+```php
+
+use WordPress\Plugin_Check\Checker\Check_Result;
+use WordPress\Plugin_Check\Checker\Check_Categories;
+use WordPress\Plugin_Check\Checker\Checks\Abstract_PHP_CodeSniffer_Check;
+use WordPress\Plugin_Check\Traits\Stable_Check;
+
+/**
+ * Check for running WordPress internationalization sniffs.
+ */
+class I18n_Usage_Check extends Abstract_PHP_CodeSniffer_Check {
```

Comment:
This example is an existing check, we should rather focus on an imaginary check. For example, keep using something like `My_Custom_Check`, and then also use another sniff from PHPCodeSniffer. This is just an example, so we can be creative.